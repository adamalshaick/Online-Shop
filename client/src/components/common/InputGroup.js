import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const InputGroup = ({ name, placeholder, value, error, type, onChange }) => {
  return (
    <div>
      <TextField
        style={{ width: "100%" }}
        label={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        type={type}
      />
      {error ? (
        <div>
          <small style={{ color: "red" }} className="entry">
            {error}
          </small>
        </div>
      ) : null}
    </div>
  );
};

InputGroup.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

InputGroup.defaultProps = {
  type: "text"
};

export default InputGroup;
