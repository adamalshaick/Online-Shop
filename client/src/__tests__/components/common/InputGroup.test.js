import React from "react";
import InputGroup from "../../../components/common/InputGroup";
import renderer from "react-test-renderer";

const wrapper = renderer.create(<InputGroup />).toJSON();

describe("render component", () => {
  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
