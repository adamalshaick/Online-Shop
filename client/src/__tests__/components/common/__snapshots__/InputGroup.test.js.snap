// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="input-group mb-3"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i />
    </span>
  </div>
  <input
    className="form-control form-control-lg"
    type="text"
  />
</div>
`;
